extension/background/intentParser.js

```diff
@@ -20,34 +20,65 @@ export function registerMatcher(intentName, matcher) {
   }
   INTENT_NAMES.push(intentName);
   INTENT_NAMES.sort();
-  INTENTS[intentName] = { matcher };
+  INTENTS[intentName] = { matcher, followUp };
 }
 
 let initialized = false;
 let phraseSet;
 
 function initialize() {
   const phrases = [];
+  let followUpPhrases;
+
   for (const name in INTENTS) {
-    const { matcher } = INTENTS[name];
+    const { matcher, followUp } = INTENTS[name];
     for (const line of splitPhraseLines(matcher)) {
       const compiled = compile(line, {
         entities: entityTypes,
         intentName: name,
       });
+
       INTENTS[name].compiledMatcher = compiled;
       phrases.push(compiled);
     }
+
+    if (followUp) {
+      const rawfollowUpPhrases = [];
+      followUp.phrase && rawfollowUpPhrases.push(followUp.phrase);
+
+      if (followUp.intent) {
+        for (const intent of followUp.intent) {
+          rawfollowUpPhrases.push({
+            label: intent.label,
+            match: INTENTS[intent.name].matcher,
+          });
+        }
+      }
+
+      followUpPhrases = rawfollowUpPhrases.flatMap(phrase => {
+        return splitPhraseLines(phrase.match).map(line => {
+          const compiled = compile(line, {
+            entities: entityTypes,
+            intentName: name,
+          });
+
+          compiled.followUpLabel = phrase.label;
+          compiled.followUpUtterance = true;
+          return compiled;
+        });
+      });
+    }
```

Comment:
This is my main concern. Since this section is already wrapped in a loop in the upper scope, compiling the follow up phrases has a complexity of n3 (yikes!)....

Reply:
I've addressed this in commit fba590fd7bfc20cb9a892e183e421da1d6ca5e52 and reduced the complexity to n^2, which is what it was before this PR